metis-indexing/src/main/java/eu/europeana/indexing/utils/RdfTierUtils.java

```diff
@@ -146,32 +148,136 @@ public static boolean hasTierCalculation(RDF rdf, Class<? extends Tier> tier) {
         .toList();
   }
 
+  @NotNull
+  private static Function<List<HasQualityAnnotation>, List<String>> mapperQualityAnnotationBodyBy(String target) {
+    return annotations -> annotations
+        .stream()
+        .map(HasQualityAnnotation::getQualityAnnotation)
+        .filter(Objects::nonNull)
+        .map(mapperQualityAnnotationByTarget(target))
+        .filter(Objects::nonNull)
+        .map(QualityAnnotation::getHasBody)
+        .filter(Objects::nonNull)
+        .map(ResourceType::getResource)
+        .filter(Objects::nonNull)
+        .toList();
+  }
+
+  @NotNull
+  private static Function<QualityAnnotation, QualityAnnotation> mapperQualityAnnotationByTarget(
+      String target) {
+    return qualityAnnotation -> qualityAnnotation
+        .getHasTargetList()
+        .stream()
+        .filter(Objects::nonNull)
+        .map(HasTarget::getResource)
+        .anyMatch(t -> t.equals(target)) ? qualityAnnotation : null;
+  }
+
+  /**
+   * Has tier calculation by target boolean.
+   *
+   * @param rdf the rdf
+   * @param tier the tier
+   * @return the boolean
+   */
+  public static boolean hasTierCalculationByTarget(RDF rdf, Class<? extends Tier> tier) {
+    List<String> tierData = extractTierDataByTarget(rdf.getAggregationList(),
+        Aggregation::getHasQualityAnnotationList,
+        rdf.getAggregationList()
+           .stream()
+           .map(Aggregation::getAbout)
+           .findFirst()
+           .orElse(null)
+    );
+    return containsTierCalculation(tier, tierData);
+  }
+
+  /**
+   * Has tier europeana calculation by target boolean.
+   *
+   * @param rdf the rdf
+   * @param tier the tier
+   * @return the boolean
+   */
+  public static boolean hasTierEuropeanaCalculationByTarget(RDF rdf, Class<? extends Tier> tier) {
+    List<String> tierEuropeanaData = extractTierData(rdf.getEuropeanaAggregationList(),
+        EuropeanaAggregationType::getHasQualityAnnotationList);
+    extractTierDataByTarget(rdf.getEuropeanaAggregationList(),
+        EuropeanaAggregationType::getHasQualityAnnotationList,
+        rdf.getEuropeanaAggregationList()
+           .stream()
+           .map(EuropeanaAggregationType::getAbout)
+           .findFirst()
+           .orElse(null)
+    );
+
+    return containsTierCalculation(tier, tierEuropeanaData);
+  }
+
   private static boolean containsTierCalculation(Class<? extends Tier> tier, List<String> tierCalculation) {
-    if (tier.isAssignableFrom(MediaTier.class)) {
+    if (tier.isInstance(MediaTier.class)) {
       return tierCalculation.stream().filter(Objects::nonNull).anyMatch(t -> t.startsWith(CONTENT_TIER_BASE_URI));
-    } else if (tier.isAssignableFrom(MetadataTier.class)) {
+    } else if (tier.isInstance(MetadataTier.class)) {
       return tierCalculation.stream().filter(Objects::nonNull).anyMatch(t -> t.startsWith(METADATA_TIER_BASE_URI));
     } else {
       return false;
     }
   }
 
+  private static String getTierBaseUri(Class<? extends Tier> tier) {
```

Comment:
I meant like:
```
  private static String getTierBaseUri(Tier tier) {
    if (tier instanceof MediaTier) {
      return CONTENT_TIER_BASE_URI;
    } else if (tier instanceof MetadataTier) {
      return METADATA_TIER_BASE_URI;
    } else {
      return "";
    }
  }
```

And where it's called you pass the Tier not it's getClass.
I think this applies in other places as well, I was wondering why it's necessary to pass `Class<? extends Tier>` instead of just Tier.

Reply:
done. refactored. using `instanceof` for type check.